ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/PeerDiscoveryController.java

```diff
@@ -391,6 +386,16 @@ private boolean addToPeerTable(final DiscoveryPeer peer) {
       notifyPeerBonded(peer, now);
     }
 
+    return addToPeerTable(peer);
+  }
+
+  public boolean addToPeerTable(final DiscoveryPeer peer) {
+    final PeerTable.AddResult result = peerTable.tryAdd(peer);
+
+    if (result.getOutcome() == PeerTable.AddResult.AddOutcome.SELF) {
+      return false;
+    }
```

Comment:
(bug) This method bypasses `peerPermissions`.

Reply:
Yes, this was missed. I will create a new PR to add this check.